content/blog/biscuit-2.0.md

```diff
@@ -0,0 +1,128 @@
+updated = 2022-14-01T09:19:42+00:00
+draft = true
+template = "blog/page.html"
+
+[taxonomies]
+authors = ["Rustaceans"]
+
+[extra]
+lead = "Biscuit 2.0 announcement"
++++
+
+We are delighted to announce the release of Biscuit at version 2.0! Biscuit is a specification
+for a cryptographically verified authorization token supporting offline delegation, and a
+language for authorization policies based on Datalog. It has been in development for 3 years
+and is already deployed in production systems.
+
+Biscuit was designed to address new challenges in building authorization in distributed systems:
+- a query going through microservices must be verifiable decentrally without giving its full
+authrization level to every node
+- API authorization increasingly has to fit a user's authorization landscape (teams, roles,
+delegation) while guaranteeing multitenant isolation
+
+As an example, [Clever Cloud](https://www.clever-cloud.com)'s Pulsar service is accessible
+with Biscuit tokens: each customer is provided a token with full access to their namespace,
+and from there new tokens can be created that are limited to producing or subscribing to
+specific topics, exactly according to application needs, without adding specific code on
+the server's side.
+
+To support those use cases, Biscuit specifies a token with public key signatures (like JWT)
+and offline attenuation (like Macaroons): any service that knows the public key can verify
+the token. And any token holder can derive a new token with more restrictions. For the
+microservices use case, that means a node can take the token it received, and create a new one
+with the least privilege before sending it to the next node.
+
+To guarantee that authorization policies execute the same way everywhere, Biscuit provides
+an authorization language based on Datalog, that all implementations must support.
+This logic language comes with benefits about execution (guaranteed to terminate) and
+serialization (small enough to be carried in the token) while keeping nearly as much
+expressiveness as SQL.
+
+You can test it right there, in your browser:
+
+{% datalog() %}
+// we receive a request to read "admin.doc"
+// The request contains a token with the following content
+user(1234);
+
+// this restricts the kind of operation to "read"
+check if operation("read");
+
+// The authorizer loads facts representing the request
+resource("admin.txt");
+operation("read");
+
+// The authorizer loads the user's rights
+right(1234, "admin.txt", "read");
+right(1234, "admin.txt", "write");
+
+// Finally, the authorizer tests policies
+// by looking for a set of facts matching them
+allow if
+  user($user_id),
+  resource($res),
+  operation($op),
+  right($user_id, $res, $op);
+{% end %}
+
+While Biscuit has great ambitions for your systems, it can be integrated right
+now without replacing the entire authorization stack: you can use it to carry
+user ids and API keys, and benefit from attenuation on the client side.
```

Comment:
maybe an example would be nice, a token carrying a session id, with stateless checks (TTL, and source ip address for instance)